src/transformState.js

```diff
@@ -5,7 +5,38 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    const { type } = action;
+
+    switch (action.type) {
+      case 'addProperties':
+        const { extraData } = action;
+
+        for (const [key, value] of Object.entries(extraData)) {
+          state[key] = value;
+        }
+        break;
```

Comment:
you can also achieve that using just[ spread operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Spread_syntax) or `Object.assign()` , that way there will be no need to creating the loop